### PR TITLE
feat: (Gen AI) Updates for Vertex `embeddings` samples

### PR DESCRIPTION
**Pull request by Valeriy-Burlaka**
## Description

* Add a new sample for the "Specify lower-dimension embeddings"  [section](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-multimodal-embeddings#low-dimension) of the `embeddings/get-multimodal-embeddings` Vertex AI [page](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-multimodal-embeddings).
* Merges all test files into one, according to the new preferred structure (i.e., having only one test file per samples dir, named as `test_*.py`)
* Fix the links to the relevant documentation sections in other `multimodal/embeddings` samples.
* Specify model versions in all `embeddings` samples
* Update the `multimodal_video_example.py` to be video-only, as per the relevant [documentation](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-multimodal-embeddings#video-modes).
* Various other fixes and improvements for `embeddings` samples (simplify function signatures, fix `TODO: developer ...` comments, add example responses, etc.)

## Checklist
- [X] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [X] **Tests** pass:   `nox -s py-3.8` and `nox -s py-3.11`
- [X] **Lint** pass:   `nox -s lint` 
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: generative_ai/embeddings/multimodal_video_example.py [hunk 1]
```diff
@@ -19,9 +19,9 @@
 
 
 def get_video_embeddings() -> MultiModalEmbeddingResponse:
```
Comment (Valeriy-Burlaka) [inline generative_ai/embeddings/multimodal_video_example.py:21]: The [documentation](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-multimodal-embeddings#vid-embedding) for this sample says:

> Use the following sample to get embeddings for video content **alone**.

I believe the goal for this sample should be to show that the user can still use the `MultiModalEmbeddingModel` class to get one type of embeddings (e.g., video-only)

File: generative_ai/embeddings/multimodal_image_text_example.py
Comment (Valeriy-Burlaka) [inline generative_ai/embeddings/multimodal_image_text_example.py:1]: Renamed the sample to reflect that it gets multimodal embeddings (image **and** text) and added correct URL to the relevant documentation [section](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-multimodal-embeddings#text-image-embedding).

File: generative_ai/embeddings/multimodal_image_text_example.py [hunk 2]
```diff
@@ -19,10 +19,10 @@
 PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
 
 
-def get_image_embeddings() -> MultiModalEmbeddingResponse:
+def get_image_text_embeddings() -> MultiModalEmbeddingResponse:
```
Comment (Valeriy-Burlaka) [inline generative_ai/embeddings/multimodal_image_text_example.py:22]: renamed for clarity and fixed the link to the documentation

File: generative_ai/embeddings/document_retrieval_example.py [hunk 3]
```diff
@@ -13,23 +13,25 @@
...
 
 from vertexai.language_models import TextEmbeddingInput, TextEmbeddingModel
 
 
 def embed_text(
-    texts: list = None,
-    task: str = "RETRIEVAL_DOCUMENT",
-    dimensionality: Optional[int] = 256,
-) -> List[List[float]]:
+    texts: list[str] | None = None,
```
Comment (msampathkumar) [inline generative_ai/embeddings/document_retrieval_example.py:22]: @Valeriy-Burlaka - Can we improve this function to have no arguments?

If need, you can define these variables inside the function itself. More details @ go/devrel-hidden-variables

File: generative_ai/embeddings/generate_embeddings_with_lower_dimension.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+# limitations under the License.
+
+import os
+
+from vertexai.vision_models import MultiModalEmbeddingResponse
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def generate_embeddings_with_lower_dimension(embedding_dimension: int = 128) -> MultiModalEmbeddingResponse:
```
Comment (msampathkumar) [inline generative_ai/embeddings/generate_embeddings_with_lower_dimension.py:22]: Same comment as before.

Comment (Valeriy-Burlaka) [reply generative_ai/embeddings/generate_embeddings_with_lower_dimension.py]: Addressed this one.

File: generative_ai/embeddings/batch_example.py [hunk 5]
```diff
@@ -28,7 +28,7 @@ def embed_text_batch() -> BatchPredictionJob:
     import vertexai
     from vertexai.preview import language_models
 
-    # TODO(developer): Uncomment and set your project ID
+    # TODO(developer): Update & uncomment line below
```
Comment (Valeriy-Burlaka) [inline generative_ai/embeddings/batch_example.py:31]: fixed `TODO: ...` comments according to the runnable-snippets-plan for Python

File: generative_ai/embeddings/document_retrieval_example.py [hunk 6]
```diff
@@ -13,32 +13,32 @@
 # limitations under the License.
 
 # [START generativeaionvertexai_embedding]
-from typing import List, Optional
+from __future__ import annotations
```
Comment (Valeriy-Burlaka) [inline generative_ai/embeddings/document_retrieval_example.py:16]: don't use imports from `typings` (it's deprecated)

File: generative_ai/embeddings/document_retrieval_example.py [hunk 7]
```diff
@@ -13,32 +13,32 @@
...
 
 from vertexai.language_models import TextEmbeddingInput, TextEmbeddingModel
 
 
-def embed_text(
```
Comment (Valeriy-Burlaka) [inline generative_ai/embeddings/document_retrieval_example.py:21]: simplify function signature

File: generative_ai/embeddings/document_retrieval_example.py [hunk 8]
```diff
@@ -13,32 +13,33 @@
...
+    dimensionality = 256
+    # The task type for embedding. Check the available tasks in the model's documentation.
+    task = "RETRIEVAL_DOCUMENT"
+
     model = TextEmbeddingModel.from_pretrained("text-embedding-004")
     inputs = [TextEmbeddingInput(text, task) for text in texts]
     kwargs = dict(output_dimensionality=dimensionality) if dimensionality else {}
     embeddings = model.get_embeddings(inputs, **kwargs)
+
+    print(embeddings)
```
Comment (Valeriy-Burlaka) [inline generative_ai/embeddings/document_retrieval_example.py:40]: missing `print`

File: generative_ai/embeddings/generate_embeddings_with_lower_dimension.py
Comment (Valeriy-Burlaka) [inline generative_ai/embeddings/generate_embeddings_with_lower_dimension.py:1]: New sample (see the PR description for links to the documentation section)

File: generative_ai/embeddings/multimodal_example.py [hunk 9]
```diff
@@ -22,19 +22,19 @@
 def get_image_video_text_embeddings() -> MultiModalEmbeddingResponse:
     """Example of how to generate multimodal embeddings from image, video, and text.
 
-    Read more @ https://cloud.google.com/vertex-ai/docs/generative-ai/embeddings/get-multimodal-embeddings#video-best-practices
+    Read more @ https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-multimodal-embeddings#img-txt-vid-embedding
```
Comment (Valeriy-Burlaka) [inline generative_ai/embeddings/multimodal_example.py:25]: bad URL (copy paste)

File: generative_ai/embeddings/test_embeddings.py
Comment (Valeriy-Burlaka) [inline generative_ai/embeddings/test_embeddings.py:1]: Merge all test files into one

File: generative_ai/embeddings/batch_example.py [hunk 10]
```diff
@@ -38,7 +38,7 @@ def embed_text_batch() -> BatchPredictionJob:
     output_uri = OUTPUT_URI
 
     textembedding_model = language_models.TextEmbeddingModel.from_pretrained(
-        "textembedding-gecko"
+        "textembedding-gecko@003"
```
[the remark this answers is not in the document]
Comment (msampathkumar) [reply generative_ai/embeddings/batch_example.py:41]: Good!

**Review by msampathkumar — APPROVED** [review]
LGTM.

**Pull request merged**